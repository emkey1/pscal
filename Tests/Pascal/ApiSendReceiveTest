program apiSendReceiveTest;

var
  ms: mstream;
  response: string;
  url: string;
  cwd: string;
begin
  cwd := getenv('PWD');
  url := 'file://' + cwd + '/Pascal/apiSendReceiveTest.data';
  ms := apiSend(url, '');
  response := apiReceive(ms);
  if pos('Example Domain', response) = 0 then
  begin
    writeln('Unexpected API response');
    halt(1);
  end;
  writeln('API send/receive test passed');
end.
