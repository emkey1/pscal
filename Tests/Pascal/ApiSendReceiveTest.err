api_send: curl_easy_perform() failed: HTTP response code said error
[Error Location] Offset: 32, Line: 32
