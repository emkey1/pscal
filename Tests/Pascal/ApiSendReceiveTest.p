program ApiSendReceiveTest;

var
  ms: mstream;
  response: string;
  url: string;
begin
  url := 'http://example.com/';
  ms := api_send(url, '');
  response := api_receive(ms);
  if pos('Example Domain', response) = 0 then
  begin
    writeln('Unexpected API response');
    halt(1);
  end;
  writeln('API send/receive test passed');
end.
