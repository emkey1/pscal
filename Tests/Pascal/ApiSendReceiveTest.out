API send/receive test passed
